kerchunk/tests/test_hdf.py

```diff
@@ -1,31 +1,50 @@
+import asyncio
 import fsspec
 import os.path as osp
 
+import fsspec.implementations
+import fsspec.implementations.reference
+
 import kerchunk.hdf
 import numpy as np
 import pytest
 import xarray as xr
 import zarr
-import h5py
+
+from packaging.version import Version
 
 from kerchunk.hdf import SingleHdf5ToZarr, has_visititems_links
 from kerchunk.combine import MultiZarrToZarr, drop
 
 here = osp.dirname(__file__)
 
 
+async def list_dir(store, path):
+    [x async for x in store.list_dir(path)]
+
+
+def create_store(test_dict: dict):
+    if Version(zarr.__version__) < Version("3.0.0.a0"):
```

Comment:
May as well make the filesystem up front to ensure it's the same for both arms (the else was missing remote_ arguments):
```
fs = fsspec.filesystem("reference", fo=test_dict, remote_protocol="s3", remote_options=so)
```
(and then pass to `RemoteStore` or call `.get_mapper()`)